tests/unit/test_table.py

```diff
@@ -1630,6 +1630,40 @@ def test_iterate(self):
 
         api_request.assert_called_once_with(method="GET", path=path, query_params={})
 
+    def test_iterate_with_cached_first_page(self):
```

Comment:
I add these tests to fix coverage.

Perhaps instead I should also be reverting the changes from https://github.com/googleapis/python-bigquery/pull/384 ?

Reply:
The alternative I'd consider is doing a flag-based feature, so users can turn on the latency changes (or flip default, and provide an opt-out while we diagnose these other cases).